tests_e2e/tests/ext_update/ext_update_failure.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import uuid
+from typing import List, Dict, Any
+
+from assertpy import fail, assert_that
+
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds, VmExtensionIdentifier
+
+
+class ExtensionUpdateFailureTest(AgentVmTest):
+    """
+    This test verifies that the agent reports correct status and code when extension operation fails during the ext update.
+    """
+
+    def run(self):
+        ssh_client: SshClient = self._context.create_ssh_client()
+        is_arm64: bool = ssh_client.get_architecture() == "aarch64"
+
+        custom_script_2_0 = VirtualMachineExtensionClient(
+            self._context.vm,
+            VmExtensionIds.CustomScript,
+            resource_name="CustomScript")
+
+        if is_arm64:
+            log.info("Will skip the update scenario, since currently there is only 1 version of CSE on ARM64")
```

Comment:
Is there an extension we can use with more than 1 version in arm64?

Reply:
This scenario is independent to distro or type. Hence, I use random(endorsed).

Reply:
I checked other extensions, they are not supported in all distros as well. Then we need write logic to skip on those. Since scenario not distro dependent, I'll skip for arm64